### PR TITLE
Devengage 1747 refactor json

### PR DESCRIPTION
**Pull request by carnellj-genesys**

### CODE REVIEW

File: genesyscloud/tfexporter/genesyscloud_resource_exporter.go [hunk 1]
```diff
@@ -246,24 +232,24 @@ func (g *GenesysCloudResourceExporter) retrieveGenesysCloudObjectInstances() dia
 	return nil
 }
 
-// buildJsonConfigMap Builds a map of all the JSON data returned for each resource
-func (g *GenesysCloudResourceExporter) buildJsonConfigMap() diag.Diagnostics {
-	log.Printf("Build JSON Map and HCL Blocks")
-	g.resourceTypeJSONMaps = make(map[string]map[string]gcloud.JsonMap)
```
Comment (carnellj-genesys) [inline genesyscloud/tfexporter/genesyscloud_resource_exporter.go:252]: This was a small nit.  We are not really dealing with a JSON map.  We are dealing with a map of different resource types.  The map is used to generate JSON or HCL so I renamed the attribute

File: genesyscloud/tfexporter/genesyscloud_resource_exporter.go [hunk 2]
```diff
@@ -316,269 +319,15 @@ func (g *GenesysCloudResourceExporter) generateOutputFiles() diag.Diagnostics {
 	return nil
 }
 
-/*****ADDED CODE HERE*****/
```
Comment (carnellj-genesys) [inline genesyscloud/tfexporter/genesyscloud_resource_exporter.go:319]: This code was moved to different files that linked closer to how they were being used.

File: genesyscloud/tfexporter/hcl_exporter.go [hunk 3]
```diff
@@ -4,44 +4,66 @@ import (
...
-	unresolvedAttrs []unresolvableAttributeInfo,
-	providerSource,
-	version,
-	filePath,
-	tfVarsFilePath string) diag.Diagnostics {
+/*
+   This file contains all of the functions used to export HCL functions.
+*/
+
+type HCLExporter struct {
```
Comment (carnellj-genesys) [inline genesyscloud/tfexporter/hcl_exporter.go:19]: move to a struct to encapsulate many of the HCL functions

File: genesyscloud/tfexporter/json_exporter.go [hunk 4]
```diff
@@ -1,35 +1,57 @@
...
 )
 
-func exportJSONConfig(
-	resourceTypeJSONMaps map[string]map[string]gcloud.JsonMap,
-	unresolvedAttrs []unresolvableAttributeInfo,
-	providerSource,
-	version,
-	filePath,
-	tfVarsFilePath string) diag.Diagnostics {
+type JsonExporter struct {
```
Comment (carnellj-genesys) [inline genesyscloud/tfexporter/json_exporter.go:14]: Moved core exporter logic to a struct

File: genesyscloud/tfexporter/tfstate_exporter.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
+package tfexporter
```
Comment (carnellj-genesys) [inline genesyscloud/tfexporter/tfstate_exporter.go:1]: All the tfstate file code is moved to its own file

**Review by charliecon — APPROVED** [review]

**Comment by charliecon** [comment]
Looks like there's a problem with go generate

**Pull request merged**